### DISCUSSION TITLE
Why Picker.prompt can't setting value?

### DISCUSSION THREAD

**Discussion by cksal0805** [ant-design/ant-design-mobile#5249] [Q&A] [answered]
Why Picker.prompt can't setting value?

I used prompt because I didn't want to define the components every time.

However, if I use prompt, the value cannot be set, so it is initialized every time.

Is there any other way?

```js
prompt: (props: Omit<PickerProps, 'value' | 'visible' | 'children'>) => Promise<PickerValue[] | null>
```

```js
<DemoBlock title='指令式调用'>
    <Button
      onClick={async () => {
        const value = await Picker.prompt({
          columns: basicColumns,
          value: [1] // <---------------------- 
        })
        Toast.show(`${value}`)
      }}
    >
      弹出 Picker
    </Button>
  </DemoBlock>
```

**Comment by awmleer** [comment]
Can you write a simple demo on[codesandbox](https://codesandbox.io/s/antd-mobile-snrxr?file=/package.json)? It'll help us to check the problem.

**Reply by cksal0805** [reply]
@awmleer You can check the code here!

https://codesandbox.io/s/tender-neumann-rlfeg0?file=/pages/picker.js

When I changed the status to "Bar" I hope status will be reflected in Picker.

**Comment by GOWxx** [comment, marked as answer]
I think this works
```typescript

        const [pickerState, setPickerState] = useState([]);

        onClick={async () => {
          await Picker.prompt({
            columns: [
              [
                { label: "Foo", value: "foo" },
                { label: "Bar", value: "bar" }
              ]
            ],
            onConfirm: (value) => setPickerState(value),
            value: pickerState
          });
        }}
```

I fork your codesandbox and make changes
```typescript
// line: 5
const [pickerState, setPickerState] = useState([]);
// line: 21
value: pickerState
```
you can check the code [here](https://codesandbox.io/s/tender-https-xfd9i2?file=/pages/picker.js)

**Reply by cksal0805** [reply]
https://user-images.githubusercontent.com/69236698/171559072-991b887d-9876-4eaf-9934-2884e77b49a4.mov

Thank you for your answer. But it didn't work out.
I don't know why this isn't provided.
What I want is to keep the value of Picker, but when I call the prompt of Picker, Picker points to the first value every time.

And I don't think I can put the value in the props of the prompt.
<img width="911" alt="스크린샷 2022-06-02 오후 2 32 10" src="https://user-images.githubusercontent.com/69236698/171559628-ecc31cd3-fc19-4ff5-91bb-0847fc337023.png">

https://mobile.ant.design/components/picker#imperative-call

**Reply by GOWxx** [reply]
I think it's possible that the codesandbox URL you visited is not mine.
try check this?
![image](https://user-images.githubusercontent.com/12870715/171601673-910b5fb3-21a0-4592-b557-15f3f4e50f91.png)

And I recorded a video below.

https://user-images.githubusercontent.com/12870715/171602124-52f80345-e9ac-47be-acdc-d667c4e47ba6.mov

**Reply by cksal0805** [reply]
Thank you so much.
It worked out thanks to you.🥹🥰
